python/ray/experimental/data/impl/compute.py

```diff
@@ -136,11 +149,10 @@ def cache_wrapper(fn: Union[CallableClass, Callable[[Any], Any]]
     if isinstance(fn, CallableClass):
 
         def _fn(item: Any) -> Any:
-            global cached_cls, cached_fn
-            if cached_fn is None or cached_cls != fn:
-                cached_cls = fn
-                cached_fn = fn()
-            return cached_fn(item)
+            if ray.data._cached_fn is None or ray.data._cached_cls != fn:
```

Comment:
Fix necessary to support re-use across pipeline datasets (otherwise the cloudpickle created separate modules effectively disabling global variable re-use).